src/c++/library/http_client.h

```diff
@@ -42,6 +42,50 @@ typedef std::map<std::string, std::string> Headers;
 /// The key-value map type to be included as URL parameters.
 typedef std::map<std::string, std::string> Parameters;
 
+// The options for authorizing and authenticating SSL/TLS connections
+struct HttpSslOptions {
+  enum CERTTYPE { CERT_PEM = 0, CERT_DER = 1 };
+  enum KEYTYPE {
+    KEY_PEM = 0,
+    KEY_DER = 1
+    // TODO: Support loading private key from crypto engine
+    // KEY_ENG = 2
+  };
+  explicit HttpSslOptions()
+      : verify_peer(1), verify_host(2), cert_type(CERTTYPE::CERT_PEM),
+        key_type(KEYTYPE::KEY_PEM)
```

Comment:
Just to confirm, this default constructor means "no SSL connection"?

Reply:
These are the default SSL settings that are used in the libcurl. I picked the defaults from libcurl library. Whether or not they are exercised depends upon what scheme is provided in the url. See here for more details: https://github.com/triton-inference-server/client/tree/tanmayv-ssl#ssltls

Reply:
Added the describing comment in the Create constructor below.

Reply:
So this is the default SSL option for libcurl, and whether to use SSL or not depends on the schema used in the URL? i.e. this option will be ignored if "http://" is used?